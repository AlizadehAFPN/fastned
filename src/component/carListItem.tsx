import React from 'react';
import { TouchableOpacity, View, Text } from 'react-native';
import FastImage from 'react-native-fast-image';
import {CarInfo } from '../Interface';
import { carListStyles } from '../Styles/carListStyles';

interface CarListItemProps {
  item: CarInfo;
  onPressCar: (id: number) => void;
}

export const CarListItem: React.FC<CarListItemProps> = ({ item, onPressCar } ) => {
  return (
    <TouchableOpacity onPress={() => onPressCar(item?.id)} style={carListStyles.carItem}>
      <View style={carListStyles.carInfoContainer}>
        <View style={carListStyles.carInfoRow}>
          <Text style={carListStyles.labelText}>Brand:</Text>
          <Text>{item?.brand}</Text>
        </View>
        <View style={carListStyles.carInfoRow}>
          <Text style={carListStyles.labelText}>Model:</Text>
          <Text style={carListStyles.carText}>{item?.model}</Text>
        </View>
        <View style={carListStyles.carInfoRow}>
          <Text style={carListStyles.labelText}>Version:</Text>
          <Text>{item?.version}</Text>
        </View>
        <View style={carListStyles.carInfoRow}>
          <Text style={carListStyles.labelText}>Category:</Text>
          <Text>{item?.category}</Text>
        </View>
      </View>
      <FastImage
        style={carListStyles.carImage}
        source={{
          uri: item.imageUrl,
          priority: FastImage.priority.normal,
        }}
        resizeMode={FastImage.resizeMode.contain}
      />
    </TouchableOpacity>
  );
};