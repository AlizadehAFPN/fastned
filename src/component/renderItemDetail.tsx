import React from 'react';
import { View, Text } from 'react-native';

import { carDetailStyles } from '../Styles/carDetailStyles';


export const renderDetail= (label:string, value:string) => (
      <View style={carDetailStyles.detailRow}>
        <Text style={carDetailStyles.detailLabel}>{label}:</Text>
        <Text style={carDetailStyles.detailValue}>{value}</Text>
      </View>
  );