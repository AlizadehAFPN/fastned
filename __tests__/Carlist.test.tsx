import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { CarListItem } from '../src/component/carListItem';

describe('CarListItem Component', () => {
  const mockItem = {
    id: 1,
    brand: 'Brand 1',
    model: 'Model 1',
    version: 'Version 1',
    category: 'Category 1',
    imageUrl: 'image-url-1',
  };

  it('renders correctly', () => {
    const { getByText, getByTestId } = render(
      <CarListItem item={mockItem} onPressCar={() => {}} />
    );

    // Check if the component renders the correct content
    expect(getByText('Brand:')).toBeTruthy();
    expect(getByText('Brand 1')).toBeTruthy();
    expect(getByText('Model:')).toBeTruthy();
    expect(getByText('Model 1')).toBeTruthy();
    expect(getByText('Version:')).toBeTruthy();
    expect(getByText('Version 1')).toBeTruthy();
    expect(getByText('Category:')).toBeTruthy();
    expect(getByText('Category 1')).toBeTruthy();

    // Check if the test IDs are correctly assigned
    expect(getByTestId('post-row-1')).toBeTruthy();
  });

  it('calls onPressCar when pressed', () => {
    const mockOnPressCar = jest.fn();
    const { getByTestId } = render(
      <CarListItem item={mockItem} onPressCar={mockOnPressCar} />
    );

    // Simulate a press on the component
    fireEvent.press(getByTestId('post-row-1'));

    // Check if onPressCar was called with the correct item ID
    expect(mockOnPressCar).toHaveBeenCalledWith(1);
  });
});
