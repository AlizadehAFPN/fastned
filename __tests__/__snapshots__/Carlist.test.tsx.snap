// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RCTScrollView
      applyWindowCorrection={[Function]}
      canChangeSize={true}
      contentContainerStyle={
        {
          "backgroundColor": undefined,
          "minHeight": 1,
          "minWidth": 1,
          "paddingLeft": 0,
          "paddingRight": 0,
        }
      }
      contentHeight={0}
      contentWidth={0}
      data={[]}
      dataProvider={
        DataProvider {
          "_data": [],
          "_firstIndexToProcess": 0,
          "_hasStableIds": false,
          "_requiresDataChangeHandling": false,
          "_size": 0,
          "getStableId": [Function],
          "rowHasChanged": [Function],
        }
      }
      disableRecycling={false}
      estimatedItemSize={100}
      extendedState={{}}
      externalScrollView={[Function]}
      finalRenderAheadOffset={250}
      forceNonDeterministicRendering={true}
      horizontal={false}
      initialOffset={0}
      initialRenderIndex={0}
      isHorizontal={false}
      layoutProvider={
        GridLayoutProviderWithProps {
          "_acceptableRelayoutDelta": 1,
          "_getHeightOrWidth": [Function],
          "_getLayoutTypeForIndex": [Function],
          "_getSpan": [Function],
          "_hasExpired": false,
          "_maxSpan": 1,
          "_setLayoutForType": [Function],
          "_tempDim": {
            "height": 0,
            "width": 0,
          },
          "averageWindow": AverageWindow {
            "currentAverage": 100,
            "currentCount": 1,
            "inputValues": [
              100,
            ],
            "nextIndex": 1,
          },
          "defaultEstimatedItemSize": 100,
          "layoutObject": {
            "size": undefined,
            "span": undefined,
          },
          "props": {
            "data": [],
            "estimatedItemSize": 100,
            "numColumns": 1,
            "renderItem": [Function],
          },
          "renderWindowInsets": {
            "height": 0,
            "width": 0,
          },
          "shouldRefreshWithAnchoring": true,
        }
      }
      maxRenderAhead={750}
      numColumns={1}
      onEndReached={[Function]}
      onEndReachedThreshold={0}
      onEndReachedThresholdRelative={0}
      onItemLayout={[Function]}
      onLayout={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onSizeChanged={[Function]}
      onVisibleIndicesChanged={[Function]}
      removeClippedSubviews={false}
      renderAheadOffset={0}
      renderAheadStep={250}
      renderContentContainer={[Function]}
      renderItem={[Function]}
      renderItemContainer={[Function]}
      rowRenderer={[Function]}
      scrollEventThrottle={16}
      scrollThrottle={16}
      scrollViewProps={
        {
          "contentContainerStyle": {
            "backgroundColor": undefined,
            "minHeight": 1,
            "minWidth": 1,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          "onLayout": [Function],
          "onScrollBeginDrag": [Function],
          "refreshControl": undefined,
          "style": {
            "minHeight": 1,
            "minWidth": 1,
          },
        }
      }
      style={
        {
          "minHeight": 1,
          "minWidth": 1,
        }
      }
      suppressBoundedSizeException={true}
      windowCorrectionConfig={
        {
          "applyToInitialOffset": true,
          "applyToItemScroll": true,
          "value": {
            "endCorrection": 0,
            "startCorrection": 0,
            "windowShift": -0,
          },
        }
      }
    >
      <View>
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              {
                "paddingLeft": 0,
                "paddingTop": 0,
              }
            }
          />
          <View
            style={
              [
                undefined,
                undefined,
              ]
            }
          />
          <AutoLayoutView
            enableInstrumentation={false}
            horizontal={false}
            onBlankAreaEvent={[Function]}
            onLayout={[Function]}
            renderAheadOffset={0}
            scrollOffset={0}
            style={
              {
                "height": 0,
                "width": 0,
              }
            }
            windowSize={0}
          />
          <CellContainer
            index={-1}
            style={
              [
                undefined,
                undefined,
              ]
            }
          />
          <View
            style={
              {
                "paddingBottom": 0,
                "paddingRight": 0,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
